subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/processing/IncrementalFiler.java

```diff
@@ -45,24 +44,17 @@
 
     @Override
     public final JavaFileObject createSourceFile(CharSequence name, Element... originatingElements) throws IOException {
-        recordGeneratedType(name, originatingElements, messager);
+        recordGeneratedType(name, originatingElements);
```

Comment:
Looks like we don't distinguish source file and class file here, which means `recordGeneratedType` will only record type name instead of file name - `Filer` will generate different files with `.java` and `.class`. Does this difference matter in current implementation?

Reply:
It doesn't, we are only interested in type names. The incremental compiler then deletes both .class and .java files with that name.